deeprvat/data/dense_gt.py

```diff
@@ -267,24 +305,71 @@ def setup_phenotypes(
             self.phenotype_df = self.phenotype_df.join(
                 sim_phenotype
             )  # TODO on = , validate = "1:1"
+        if sample_file is not None:
+            logger.info(f"Using samples from sample file {sample_file}")
+            with open(sample_file, "rb") as f:
+                samples_to_keep = pickle.load(f)
+            samples_to_keep = np.array(samples_to_keep)
+            logger.info(f"Number of samples in sample file: {len(samples_to_keep)}")
+            samples_to_keep = np.array(
+                list(set(samples_to_keep).intersection(set(samples_phenotype_df)))
+            )
+            logger.info(
+                f"Number of samples in sample file and in phenotype_df: {len(samples_to_keep)}"
+            )
+        else:
+            logger.info("Using all samples in phenotyp df")
+            samples_to_keep = copy.deepcopy(samples_phenotype_df)
 
+        logger.info("Removing samples that are not in genotype file")
+
+        samples_to_keep = np.array(
```

Comment:
I would add a warning that some samples from sample file were not found in the data

Reply:
added the warning